WalletWasabi/Helpers/EnvironmentHelpers.cs

```diff
@@ -32,9 +32,16 @@ internal static int ProcessorCount
 			}
 		}
 
+		private static string DataDirBacking = null;
+
 		// Do not change the output of this function. Backwards compatibility depends on it.
 		public static string GetDataDir(string appName)
 		{
+			if (DataDirBacking != null)
```

Comment:
I don't know, returning a value ignoring the parameter looks weird to me. Imagine I call it twice with different appNames.

Reply:
True, this is incorrect.

Reply:
Fixed with a dictionary.